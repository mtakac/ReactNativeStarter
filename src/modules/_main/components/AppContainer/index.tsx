import React from 'react';
import { View } from 'react-native';

import styles from './styles';

export type AppContainerProps = {
  children: React.ReactNode;
};

const AppContainer: React.FC<AppContainerProps> = (props) => {
  return <View style={styles.appContainer}>{props.children}</View>;
};

export default React.memo(AppContainer);
