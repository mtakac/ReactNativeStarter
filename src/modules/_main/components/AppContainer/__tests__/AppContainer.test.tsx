import { cleanup, render, toJSON } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';

import { store } from '~redux';

import AppContainer from '../index';

const children = 'Children';

const Providers: React.FC = (props) => <Provider store={store}>{props.children}</Provider>;

const renderAppContainer = () =>
  render(<AppContainer>{children}</AppContainer>, {
    wrapper: Providers,
  });

describe('main', () => {
  describe('<AppContainer />', () => {
    it('renders correctly', () => {
      const { container } = renderAppContainer();

      expect(toJSON(container)).toMatchSnapshot();
      cleanup();
    });
  });
});
