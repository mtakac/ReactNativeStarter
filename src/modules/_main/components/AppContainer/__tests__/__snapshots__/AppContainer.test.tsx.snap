// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main <AppContainer /> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    Children
  </View>
</View>
`;
